schema-engine/sql-introspection-tests/tests/multi_schema/mysql.rs

```diff
@@ -0,0 +1,806 @@
+
+    api.database().raw_cmd(&create_schema).await?;
+    api.database().raw_cmd(&create_table).await?;
+
+    let input = indoc! {r#"
+        model A {
+          id Int @id
+          B  B[]
+
+          @@schema("first_m6")
+        }
+
+        model B {
+          id Int  @id
+          fk Int?
+          A  A?   @relation(fields: [fk], references: [id], onDelete: Restrict, onUpdate: Restrict)
+
+          @@schema("first_m6")
+        }
+    "#};
+
+    let expected = expect![[r#"
+        model A {
+          id Int @id
+          B  B[]
+
+          @@schema("first_m6")
+        }
+
+        model B {
+          id Int  @id
+          fk Int?
+          A  A?   @relation(fields: [fk], references: [id], onDelete: Restrict, onUpdate: Restrict, map: "fk_tbl")
+
+          @@index([fk], map: "fk_tbl")
+          @@schema("second_m6")
+        }
+    "#]];
+
+    api.expect_re_introspected_datamodel(input, expected).await;
+
+    Ok(())
+}
+
+#[test_connector(tags(Mysql), preview_features("multiSchema"), namespaces("first_m7", "second_m7"))]
+async fn multiple_schemas_w_cross_schema_fks_w_duplicate_names_are_introspected(api: &mut TestApi) -> TestResult {
+    let schema_name = "first_m7";
+    let other_name = "second_m7";
+    let create_schema = format!("CREATE SCHEMA `{schema_name}`",);
+    let create_table = format!("CREATE TABLE `{schema_name}`.`A` (id INT PRIMARY KEY)",);
+    //Todo
+    api.database().raw_cmd(&create_schema).await?;
+    api.database().raw_cmd(&create_table).await?;
+
+    let create_schema = format!("CREATE SCHEMA `{other_name}`",);
+    let create_table =
+        format!("CREATE TABLE `{other_name}`.`A` (id INT PRIMARY KEY, fk INT, CONSTRAINT fk_tbl FOREIGN KEY (fk) REFERENCES `{schema_name}`.`A`(`id`) ON DELETE RESTRICT ON UPDATE RESTRICT)",);
+
+    api.database().raw_cmd(&create_schema).await?;
+    api.database().raw_cmd(&create_table).await?;
+
+    let expected = expect![[r#"
+        model first_m7_A {
+          id Int           @id
+          A  second_m7_A[]
+
+          @@map("A")
+          @@schema("first_m7")
+        }
+
+        model second_m7_A {
+          id Int         @id
+          fk Int?
+          A  first_m7_A? @relation(fields: [fk], references: [id], onDelete: Restrict, onUpdate: Restrict, map: "fk_tbl")
+
+          @@index([fk], map: "fk_tbl")
+          @@map("A")
+          @@schema("second_m7")
+        }
+    "#]];
+
+    let result = api.introspect_dml().await?;
+    expected.assert_eq(&result);
+
+    Ok(())
+}
+
+#[test_connector(tags(Mysql), preview_features("multiSchema"), namespaces("first_m8", "second_m8"))]
+async fn multiple_schemas_w_enums_are_introspected(api: &mut TestApi) -> TestResult {
+    let schema_name = "first_m8";
+    let other_name = "second_m8";
+    let sql = format! {
+        r#"
+            CREATE SCHEMA `{schema_name}`;
+            CREATE TABLE `{schema_name}`.`HappyPerson` (mood ENUM ('happy') PRIMARY KEY);
```

Comment:
ℹ️ Enum behavior is also a bit different as enums are no dedicated type in MySQL but are bound to the table+column. Had to make basically no modifications to the implemented logic to make this work so it doe snot work differently than regular mysql enum inspection so far.